spec/filters/aggregate_spec.rb

```diff
@@ -180,6 +180,20 @@
       end
     end
 
+    describe "timeout defined on the filter" do
+      it "event is not removed if not expired" do
+        entries = @end_filter.flush()
+        expect(aggregate_maps.size).to eq(1)
+        expect(entries).to be_empty
+      end
+      it "event is removed and creates a new event" do
+        sleep(2)
+        entries = @end_filter.flush()
+        expect(aggregate_maps).to be_empty
+        expect(entries.size).to eq(1)
+      end
+    end
+
   end
```

Comment:
Finally, it would be nice to update test "push_previous_map_as_event option is defined, " so that it verify timeout_code behaviour in `filter` method.